substrate/frame/support/src/lib.rs

```diff
@@ -1019,1222 +909,358 @@ pub mod pallet_prelude {
+	/// [`ValidateUnsigned`](frame_support::pallet_prelude::ValidateUnsigned) for
+	/// type `Pallet<T>`, and some optional where clause.
+	///
+	/// NOTE: There is also the `sp_runtime::traits::SignedExtension` trait that can be used to
+	/// add some specific logic for transaction validation.
+	///
+	/// ## Macro expansion
+	///
+	/// The macro currently makes no use of this information, but it might use this information
+	/// in the future to give information directly to [`frame_support::construct_runtime`].
+	pub use frame_support_procedural::validate_unsigned;
+
+	/// The `#[pallet::type_value]` attribute lets you define a struct implementing the
+	/// [`Get`](frame_support::traits::Get) trait to ease the use of storage types. This
+	/// attribute is meant to be used alongside [`#[pallet::storage]`](`macro@storage`) to
+	/// define a storage's default value. This attribute can be used multiple times.
+	///
+	/// Item must be defined as:
+	///
+	/// ```ignore
+	/// #[pallet::type_value]
+	/// fn $MyDefaultName<$some_generic>() -> $default_type $optional_where_clause { $expr }
+	/// ```
+	///
+	/// I.e.: a function definition with generics none or `T: Config` and a returned type.
+	///
+	/// E.g.:
+	///
+	/// ```ignore
+	/// #[pallet::type_value]
+	/// fn MyDefault<T: Config>() -> T::Balance { 3.into() }
+	/// ```
+	///
+	/// ## Macro expansion
+	///
+	/// The macro renames the function to some internal name, generates a struct with the
+	/// original name of the function and its generic, and implements `Get<$ReturnType>` by
+	/// calling the user defined function.
+	pub use frame_support_procedural::type_value;
+
+	/// Because the `pallet::pallet` macro implements
+	/// [`GetStorageVersion`](frame_support::traits::GetStorageVersion), the current storage
+	/// version needs to be communicated to the macro. This can be done by using the
+	/// `pallet::storage_version` attribute:
+	///
+	/// ```ignore
+	/// const STORAGE_VERSION: StorageVersion = StorageVersion::new(5);
+	///
+	/// #[pallet::pallet]
+	/// #[pallet::storage_version(STORAGE_VERSION)]
+	/// pub struct Pallet<T>(_);
+	/// ```
+	///
+	/// If not present, the current storage version is set to the default value.
+	pub use frame_support_procedural::storage_version;
+
+	/// The `#[pallet::hooks]` attribute allows you to specify a `Hooks` implementation for
+	/// `Pallet` that specifies pallet-specific logic.
+	///
+	/// The item the attribute attaches to must be defined as follows:
+	/// ```ignore
+	/// #[pallet::hooks]
+	/// impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> $optional_where_clause {
+	///     ...
+	/// }
+	/// ```
+	/// I.e. a regular trait implementation with generic bound: `T: Config`, for the trait
+	/// `Hooks<BlockNumberFor<T>>` (they are defined in preludes), for the type `Pallet<T>` and
+	/// with an optional where clause.
+	///
+	/// If no `#[pallet::hooks]` exists, then the following default implementation is
+	/// automatically generated:
+	/// ```ignore
```

Comment:
I was kinda hoping that we would not have any `ignore`s in this file where possible.. most of these can be a real pallet with a half dozen LoCs. wdyt? feel free to push back.